src/actions/files.js

```diff
@@ -36,3 +36,48 @@ export function getFilesForPost(teamId, channelId, postId) {
         ]), getState);
     };
 }
+
+const uploadFileRequests = {};
+export function uploadFile(teamId, channelId, fileFormData, formBoundary, rootId, requestId) {
```

Comment:
Ok, I think that's fine for now but I think in the future we should look at passing back the actual request ID generated by the driver (which can also be used to actually cancel the upload). I'm not familiar enough with this new fangled `await` and `dispatch` stuff but I think we'll need a new pattern for being to return a value while also returning the `async` to wait on